scenarios.md

```diff
@@ -0,0 +1,199 @@
+  * [IBM Cloud Container Registry (icr)][icr]
+* Registry Vendors, Projects & Products
+  * [Docker Trusted Registry][docker-dtr]
+  * [Harbor][harbor]
+  * [JFrog Artifactory][jfrog]
+* Artifact Types
+  * Container Images - OCI Org
+  * Helm Charts
+  * Singularity
+  * Operator Bundles
+
+## Scenarios
+
+### Scenario #1: Local Build, Sign, Validate
+
+Prior to doing any deployment, a developer can test the: build, sign, validate scenario.
+
+1. Locally build a container image using a non-registry specific `name:tag`, such as:  
+  `$docker build hello-world:1`
+1. Locally sign `hello-world:1` 
+2. Run the image in a local environment which was configured to only support signed artifacts. *(**note** - details on the usability is deferred from this requirement. This could be a local docker host, a local kind or minikube instance configured to only run signed images)*  
+  `$ docker run hello-world:1`
+
+**Implications of this requirement:**
+
+- The local developer has access to signing keys
+- The local environment has a policy by which it states the set of keys it accepts.
+- The signing and validation of artifacts does not require a registry.
+- The signature used for validation may be hosted in a registry, or other accessible location.
+- The lack of a registry name does not infer docker.io as the default registry. This does imply changes to either the docker client, the notary client or both.
+
+### Scenario #2: Sign, Rename, Push, Validate
+
+Once the developer has locally validated the build, sign, validate scenario, they will likely want to push the artifact to a registry used for deployment.
+
+1. Locally build and sign an artifact, such as a container image
+1. Rename the artifact to reflect the registry it will be pushed to:  
+  `$ docker tag hello-world:1 myregistry.domain.com/marketing/web:1`
+  `$docker push myregistry.domain.com/marketing/web:1`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$kubectl apply -f kubedeploy.yaml`
+  > **Note:** the above example presumes `kubedeploy.yaml` references the myregistry.domain.com/marketing/web:1 image
+
+**Implications of this requirement:**
+
+- Signatures can be validated based on the tag referenced, however the signature is not tied to a specific name. The artifact can be renamed without invalidating the signature.
+- This does not preclude tag locking scenarios, where a registry can provide users with the ability to lock a tag to a specific. If a tag is updated with another signed artifact, the tag is considered signed and follows the signature validation rules of the new signature. For example, if a base image `fx:1.0` is signed, and rebuilt with a patched version, the updated `fx:1.0` is new content, and a valid scenario.
+- If the tag for `fx:1.0` is redirected to another unsigned manifest, the signature validation of `fx:1.0` will fail as it's no longer signed. This is no different than signed binaries being updated or replaced on a users local computer.
+
```

Comment:
In addition to Scenario 2 ("Sign, Rename, Push, Validate"), I think it might be worth adding a scenario involving exporting and importing rather than pushing and pulling.  Scenario 1 ("Local Build, Sign, Validate) implies that signatures are not expected to be tied to a registry and don't need to be pushed to a registry in order for validation to occur, so it might be worth going further and specifying that the signature can be exported and imported independently of a registry as well.

Reply:
I struggle with this one. On one hand, I think we'd like to enable pull-->tag-->push to "just work", including any signatures that move with it.
On the other hand, we do need a general export/import API. If we solve the above, the export/import is more for bulk scenarios. 
Or, are you referring to something else, such as exporting/importing keys?

Reply:
> I think we'd like to enable pull-->tag-->push to "just work", including any signatures that move with it.

Agreed

> On the other hand, we do need a general export/import API. If we solve the above, the export/import is more for bulk scenarios.

I don't know that we need to define an API.  The image spec already specifies the layout of a valid image, and container runtimes like containerd are already capable of writing that to disk as a single, exported file.

Import/export has use-cases beyond bulk scenarios.  It's useful for things like backup, including to offline storage like optical media.  It's also useful for providing a mechanism to transport images into an air-gapped system.  I think we probably want signatures to go along for the ride when the images are exported.

>  Or, are you referring to something else, such as exporting/importing keys?

That's another good point.  Key management and distribution isn't really covered in this scenario doc so far but we'll need to either declare it out-of-scope or cover it as well.  (I don't think we need to figure this out before merging this doc, however.)